js/src/api/transport/http/http.js

```diff
@@ -46,19 +46,33 @@ export default class Http extends JsonRpcBase {
     };
   }
 
+  _setConnected () {
```

Comment:
I think these should have been in the JsonRpc base - so both implementations are consistent with the flags they set and the events they emit (at least the base flags, not the internal extensions).